compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -63,10 +62,25 @@ func convertRHELComponents(rc *database.RHELv2Components) []*scannerV1.RHELCompo
 		log.Warn("No RHEL packages found in scan result")
 		return nil
 	}
+
+	seenEntries := make(map[int]*database.RHELv2Package)
```

Comment:
How about indexing by name+version+module+arch while the value is the ID?

```suggestion
	seenEntries := make(map[string]int)
```

This would allow us to search for duplicates in O(1).

We could go further and not care about the ID in the map as we use the index anyway - this could lead to `make(map[string]struct{})`. As this makes it practically a set, I think there was already a set implementation available somewhere in our code.

Reply:
I agree with @vikin91 suggestion. Going even further, is there a reason that prevents us from using only one map, map[string]*scannerV1.RHELComponent? In the return, we would return all the values of the map.

Although we plan to use the index as the ID, we don't care if the ID reflects the order in the final return value, do we?

I would also rename this method to `convertAndDedupRHELComponents`.

Reply:
Both are great ideas, I like them a lot.

It is true that we are not interested in the order of items, only that the IDs are unique.
The uniqueness check here is more of a sanity safeguard than an actual dedup, as we don't expect to encounter any duplicates in an inventory run.

That said, I've switched to a `map[string]*RHELComponent` and return the values of it as a slice.